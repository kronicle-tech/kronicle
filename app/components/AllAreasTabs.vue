<template>
  <div class="my-3">
    <b-button to="/all-areas">Areas</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
