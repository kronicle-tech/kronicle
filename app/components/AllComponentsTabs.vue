<template>
  <div class="my-3">
    <b-button class="my-1" :to="{ name: 'all-components' }">
      Components
    </b-button>
    <b-button
      :to="{ name: 'all-components-tests', query: { testOutcome: 'fail' } }"
    >
      Tests
    </b-button>
    <b-button
      class="my-1"
      :to="{ name: 'all-components-cross-functional-requirements' }"
    >
      Cross Functional Requirements
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-tech-debts' }">
      Tech Debts
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-openapi-specs' }">
      OpenAPI Specs
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-graphql-schemas' }">
      GraphQL Schemas
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-key-software' }">
      Key Software
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-repos' }">
      Repos
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-errors' }">
      Errors
    </b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
