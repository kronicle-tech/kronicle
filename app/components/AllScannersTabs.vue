<template>
  <div class="my-3">
    <b-button :to="{ name: 'all-scanners' }">Scanners</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
