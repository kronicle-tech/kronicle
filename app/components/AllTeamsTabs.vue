<template>
  <div class="my-3">
    <b-button to="/all-teams">Teams</b-button>
    <b-button to="/all-teams/components">Components</b-button>
    <b-button to="/all-teams/tech-debts">Tech Debts</b-button>
    <b-button to="/all-teams/tests">Tests</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
