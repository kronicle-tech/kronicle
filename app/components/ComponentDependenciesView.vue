<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col class="graph">
          <ComponentDependencyGraph
            id="component-dependency-graph"
            :dependencies="dependencies"
            dependency-type="scope-related"
            :zoom="zoom"
            :selected-component-id="selectedComponentId"
            :scoped-component-ids="filteredComponentIds"
            :fixed-scope="true"
            :scope-related-radius="parseInt(selectedScopeRelatedRadius, 10)"
            @networkChange="networkChange"
            @selectedNodeChange="selectedNodeChange"
          />
        </b-col>
        <b-col md="3">
          <b-alert show="60" dismissible variant="info">
            Hover over or click a component's dot in the dependencies diagram to see
            more information about it
          </b-alert>

          <ComponentPanel v-if="selectedComponent" :component="selectedComponent" />

          <div v-else class="my-3">
            <ComponentFilters :components="components" />

            <b-form-group
              label-cols="6"
              label-size="sm"
              label="Radius:"
              label-for="graph-scope-related-radius"
              class="my-3 mr-2"
            >
              <b-form-select
                id="graph-scope-related-radius"
                v-model="selectedScopeRelatedRadius"
                :options="scopeRelatedRadiusOptions"
                size="sm"
              />
            </b-form-group>

            <b-form-group
              label-cols="6"
              label-size="sm"
              label="Zoom:"
              label-for="graph-zoom"
              class="my-3 mr-2"
            >
              <b-form-select
                id="graph-zoom"
                v-model="zoom"
                :options="zoomOptions"
                size="sm"
              />
            </b-form-group>

            <b-form-checkbox
              id="detailed-dependencies"
              v-model="detailed"
              :value="true"
              :unchecked-value="false"
            >
              Detailed dependencies
            </b-form-checkbox>
          </div>
        </b-col>
      </b-row>
    </b-container>

    <b-container fluid>
      <b-row>
        <b-col>
          <div
            v-if="excludedComponents && excludedComponents.length > 0"
            class="excluded-components my-3"
          >
            <h2>Excluded Components</h2>
            <ComponentTable :components="excludedComponents" />
          </div>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<style scoped>
.graph {
  overflow-x: scroll;
  scrollbar-color: #444 #111;
}
</style>

<script lang="ts">
import Vue, { PropType } from 'vue'
import {
  BAlert,
  BCol,
  BContainer,
  BFormCheckbox,
  BFormGroup,
  BFormSelect,
  BRow
} from 'bootstrap-vue'
import {
  Component,
  SummaryComponentDependencies,
  SummaryComponentDependencyNode,
  SummarySubComponentDependencies,
  SummarySubComponentDependencyNode,
} from '~/types/kronicle-service'
import { Network } from '~/types/component-dependency-graph'
import { intRange } from '~/src/arrayUtils'
import ComponentDependencyGraph from '~/components/ComponentDependencyGraph.vue'
import ComponentPanel from '~/components/ComponentPanel.vue'
import ComponentTable from '~/components/ComponentTable.vue'
import ComponentFilters from '~/components/ComponentFilters.vue'

interface Option {
  value: string | undefined
  text: string
}

export default Vue.extend({
  components: {
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-form-checkbox': BFormCheckbox,
    'b-form-group': BFormGroup,
    'b-form-select': BFormSelect,
    'b-row': BRow,
    ComponentDependencyGraph,
    ComponentFilters,
    ComponentPanel,
    ComponentTable,
  },
  props: {
    components: {
      type: Array as PropType<Component[]>,
      required: true,
    },
    componentDependencies: {
      type: Object as PropType<SummaryComponentDependencies>,
      required: true,
    },
    subComponentDependencies: {
      type: Object as PropType<SummarySubComponentDependencies>,
      required: true,
    },
    allComponents: {
      type: Array as PropType<Component[]>,
      required: true,
    },
    selectedComponentId: {
      type: String,
      default: undefined,
    },
    scopeRelatedRadius: {
      type: Number,
      default: 1,
    },
  },
  data() {
    return {
      selectedNode: undefined as
        | SummaryComponentDependencyNode
        | SummarySubComponentDependencyNode
        | undefined,
      selectedComponent: undefined as Component | undefined,
      selectedScopeRelatedRadius: this.scopeRelatedRadius,
      zoom: 100,
      detailed: false,
      network: undefined as Network | undefined,
    }
  },
  computed: {
    filteredComponentIds(): Component[] {
      return this.$store.state.componentFilters.filteredComponentIds
    },
    dependencies():
      | SummaryComponentDependencies
      | SummarySubComponentDependencies {
      return this.detailed
        ? this.subComponentDependencies
        : this.componentDependencies
    },
    scopeRelatedRadiusOptions(): Option[] {
      return intRange(0, 11).map((value) => {
        return {
          value: value.toString(),
          text: value.toString(),
        }
      })
    },
    zoomOptions(): Option[] {
      const zoomOptions = [25, 50, 75, 100, 125, 150, 200, 400]
      return zoomOptions.map((zoomOption) => {
        return {
          value: zoomOption.toString(),
          text: `${zoomOption}%`,
        }
      })
    },
    excludedComponents(): Component[] | undefined {
      if (!this.network) {
        return undefined
      }

      const componentIds = this.network.nodes.map((node) => {
        return node.node.componentId
      })

      return this.components.filter((component) => {
        return !componentIds.includes(component.id)
      })
    },
  },
  methods: {
    findComponent(id: string): Component | undefined {
      return this.allComponents.find((component) => component.id === id)
    },
    networkChange(network: Network): void {
      this.network = network
    },
    selectedNodeChange(
      _: Event,
      node:
        | SummaryComponentDependencyNode
        | SummarySubComponentDependencyNode
        | undefined
    ): void {
      if (node) {
        this.selectedNode = node
        this.selectedComponent = this.findComponent(node.componentId)
      } else {
        this.selectedNode = undefined
        this.selectedComponent = undefined
      }
    },
  },
})
</script>
