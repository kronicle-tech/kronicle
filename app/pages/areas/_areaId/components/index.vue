<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <h1 class="text-info my-3">{{ area.name }} Area - Components</h1>

          <AreaTabs :area-id="area.id" />
        </b-col>
      </b-row>
    </b-container>

    <ComponentsView :components="area.components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BCol, BContainer, BRow} from "bootstrap-vue";
import { Area } from '~/types/kronicle-service'
import AreaTabs from '~/components/AreaTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AreaTabs,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const area = await fetch(
      `${$config.serviceBaseUrl}/v1/areas/${route.params.areaId}?fields=area(id,name,components(id,name,typeId,description,tags,teams,platformId))`
    )
      .then((res) => res.json())
      .then((json) => json.area as Area)

    store.commit('componentFilters/initialize', {
      components: area.components,
      route,
    })

    return {
      area,
    }
  },
  data() {
    return {
      area: {} as Area,
    }
  },
  head(): MetaInfo {
    return {
      title: `Kronicle - ${this.area.name} Area - Components`,
    }
  },
})
</script>
