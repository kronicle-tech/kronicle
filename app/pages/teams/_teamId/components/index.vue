<template>
  <div class="m-3">
    <h1 class="text-info my-3">{{ team.name }} - Components</h1>

    <TeamTabs :team-id="team.id" />

    <ComponentsView :components="team.components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import { Team } from '~/types/kronicle-service'
import ComponentsView from '~/components/ComponentsView.vue'
import TeamTabs from '~/components/TeamTabs.vue'
import { NuxtError } from '~/src/nuxtError'

export default Vue.extend({
  components: {
    ComponentsView,
    TeamTabs,
  },
  async asyncData({ $config, route, store }) {
    const team = await fetch(
      `${$config.serviceBaseUrl}/v1/teams/${route.params.teamId}?fields=team(id,name,components(id,name,discovered,type,description,tags,teams,platformId))`
    )
      .then((res) => res.json())
      .then((json) => json.team as Team | undefined)

    if (!team) {
      throw new NuxtError('Team not found', 404)
    }

    store.commit('componentFilters/initialize', {
      components: team.components,
      route,
    })

    return {
      team,
    }
  },
  data() {
    return {
      team: {} as Team,
    }
  },
  head(): MetaInfo {
    return {
      title: `Kronicle - ${this.team.name} - Components`,
    }
  },
})
</script>
