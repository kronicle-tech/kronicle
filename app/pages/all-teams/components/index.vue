<template>
  <div class="m-3">
    <AllTeamsTabs />

    <b-card-group v-for="(row, rowIndex) in rows" :key="rowIndex" deck>
      <b-card
        v-for="(item, itemIndex) in row"
        :key="itemIndex"
        bg-variant="dark"
        :header="item.team.name"
        text-variant="white"
        class="my-3"
      >
        <b-card
          v-for="component in item.components"
          :key="component.id"
          no-body
          class="mt-1"
        >
          <b-card-header header-tag="header" class="p-1" role="tab">
            <b-button
              v-b-toggle="`accordion-${rowIndex}-${itemIndex}-${component.id}`"
              block
              variant="info"
              >{{ component.name }}</b-button
            >
          </b-card-header>
          <b-collapse
            :id="`accordion-${rowIndex}-${itemIndex}-${component.id}`"
            visible
            :accordion="`accordion-${rowIndex}-${itemIndex}`"
            role="tabpanel"
          >
            <b-card-body>
              <ComponentPanel :component="component" />
            </b-card-body>
          </b-collapse>
        </b-card>
      </b-card>
    </b-card-group>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import {
  BButton,
  BCard,
  BCardBody,
  BCardGroup,
  BCardHeader,
  BCollapse,
  VBToggle,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component, Team } from '~/types/kronicle-service'
import AllTeamsTabs from '~/components/AllTeamsTabs.vue'
import ComponentPanel from '~/components/ComponentPanel.vue'

interface Item {
  team: Team
  components: Component[]
}

export default Vue.extend({
  components: {
    AllTeamsTabs,
    'b-button': BButton,
    'b-card': BCard,
    'b-card-body': BCardBody,
    'b-card-group': BCardGroup,
    'b-card-header': BCardHeader,
    'b-collapse': BCollapse,
    ComponentPanel,
  },
  directives: {
    'b-toggle': VBToggle,
  },
  async asyncData({ $config }) {
    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id,name)`
    )
      .then((res) => res.json())
      .then((json) => json.teams)

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,teams,tags,description,notes,responsibilities,techDebts,openApiSpecs,links)`
    )
      .then((res) => res.json())
      .then((json) => json.components)

    return {
      teams,
      components,
    }
  },
  data() {
    return {
      teams: [] as Team[],
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Teams',
    }
  },
  computed: {
    rows() {
      const that = this
      const rows = [] as Array<Array<Item>>
      const teamCount = this.teams.length
      let startIndex = 0
      while (startIndex < teamCount - 1) {
        const endIndex = Math.min(startIndex + 3, teamCount)
        const row = that.teams.slice(startIndex, endIndex).map(
          (team) =>
            ({
              team,
              components: getComponents(team),
            } as Item)
        )
        rows.push(row)
        startIndex = endIndex
      }
      return rows

      function getComponents(team: Team) {
        return that.components.filter((component) => {
          return component.teams.some((componentTeam) => {
            return (
              componentTeam.type !== 'previous' &&
              componentTeam.teamId === team.id
            )
          })
        })
      }
    },
  },
})
</script>
