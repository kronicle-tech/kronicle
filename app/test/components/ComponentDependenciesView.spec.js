import ComponentDependenciesView from '@/components/ComponentDependenciesView'
import { createViewComponentWrapper } from '~/test/components/viewUtils'
import { createComponent } from '~/test/testDataUtils'

function createDependency(sourceIndex, targetIndex) {
  return {
    sourceIndex,
    targetIndex,
    relatedIndexes: [],
    manual: false,
    sampleSize: 1,
    startTimestamp: '2021-01-01T00:00:00.000Z',
    endTimestamp: '2021-01-01T00:00:01.000Z',
    duration: {
      min: 1_000_000,
      max: 1_000_000,
      p50: 1_000_000,
      p90: 1_000_000,
      p99: 1_000_000,
      p99Point9: 1_000_000,
    },
  }
}

function createComponentNode(nodeNumber) {
  return {
    componentId: `test-component-id-${nodeNumber}`,
  }
}

describe('ComponentDependenciesView', () => {
  let propsData
  let wrapper

  async function createWrapper() {
    wrapper = await createViewComponentWrapper(ComponentDependenciesView, {
      propsData,
    })
  }

  function expectNodeCount(count) {
    expect(wrapper.findAll('.node')).toHaveLength(count)
  }

  function expectDependencyCount(count) {
    expect(wrapper.findAll('.dependency')).toHaveLength(count)
  }

  function expectNodeClasses(dependencies) {
    expect(wrapper.findAll('.node')).toHaveLength(dependencies.length)
    dependencies.forEach((node) =>
      expect(
        wrapper
          .get(`#component-dependency-graph-node-${node.nodeIndex}`)
          .classes()
      ).toContain(node.nodeClass)
    )
  }

  function expectDependencyClasses(dependencies) {
    expect(wrapper.findAll('.dependency')).toHaveLength(dependencies.length)
    dependencies.forEach((dependency) =>
      expect(
        wrapper
          .get(
            `#component-dependency-graph-dependency-${dependency.dependencyIndex}`
          )
          .classes()
      ).toContain(dependency.dependencyClass)
    )
  }

  beforeEach(() => {
    propsData = {
      diagram: {
        nodes: [],
        edges: [],
      },
      allComponents: [],
    }
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when components prop is set to an empty array', () => {
    beforeEach(() => {
      propsData.components = []
    })

    test('renders no dependencies', async () => {
      await createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when components prop contains multiple components', () => {
    beforeEach(() => {
      propsData.components = [
        createComponent({ componentNumber: 1, platformNumber: 1 }),
        createComponent({ componentNumber: 2, platformNumber: 2 }),
        createComponent({ componentNumber: 3, platformNumber: 3 }),
        createComponent({ componentNumber: 4, platformNumber: 4 }),
      ]
      propsData.allComponents.push(propsData.components)
      propsData.allComponents.push(
        createComponent({ componentNumber: 5, platformNumber: 5 })
      )
    })

    describe('and diagram prop contains an empty edges array', () => {
      test('renders no connections', async () => {
        await createWrapper()
        expect(wrapper.html()).toMatchSnapshot()
        expectNodeCount(0)
        expectDependencyCount(0)
      })
    })

    describe('and diagram prop contains multiple edges', () => {
      beforeEach(() => {
        propsData.diagram = {
          nodes: [
            createComponentNode(1),
            createComponentNode(2),
            createComponentNode(3),
            createComponentNode(4),
          ],
          edges: [
            createDependency(undefined, 0),
            createDependency(0, 1),
            createDependency(1, 2),
            createDependency(1, 3),
          ],
        }
      })

      test('renders the connections', async () => {
        await createWrapper()
        expect(wrapper.html()).toMatchSnapshot()
        expectNodeCount(4)
        expectDependencyCount(3)
      })

      describe('and first platform is checked in platform filter', () => {
        beforeEach(async () => {
          await createWrapper()
          await wrapper.get('input[value="test-platform-id-1"]').setChecked()
        })

        test('renders the connections directly related to the first component using the first platform', () => {
          expect(wrapper.html()).toMatchSnapshot()
          expectNodeCount(2)
          expectDependencyCount(1)
        })
      })

      describe('when a node is clicked', () => {
        beforeEach(async () => {
          await createWrapper()
          await wrapper
            .get('#component-dependency-graph-node-0')
            .trigger('click')
        })

        test('renders the connention directly related to the first node with the style of a direct dependency', () => {
          expect(wrapper.html()).toMatchSnapshot()
          expectNodeClasses([
            { nodeIndex: 0, nodeClass: 'selected-node' },
            { nodeIndex: 1, nodeClass: 'scoped-node' },
            { nodeIndex: 2, nodeClass: 'scoped-node' },
            { nodeIndex: 3, nodeClass: 'scoped-node' },
          ])
          expectDependencyClasses([
            { dependencyIndex: 1, dependencyClass: 'direct-dependency' },
            { dependencyIndex: 2, dependencyClass: 'scoped-dependency' },
            { dependencyIndex: 3, dependencyClass: 'scoped-dependency' },
          ])
        })
      })

      describe('and the mouse hovers over the first node', () => {
        beforeEach(async () => {
          await createWrapper()
          await wrapper
            .get('#component-dependency-graph-node-0')
            .trigger('mouseover')
        })

        test('renders the connection directly related to the first node with the style of a direct connection', () => {
          expect(wrapper.html()).toMatchSnapshot()
          expectNodeClasses([
            { nodeIndex: 0, nodeClass: 'selected-node' },
            { nodeIndex: 1, nodeClass: 'scoped-node' },
            { nodeIndex: 2, nodeClass: 'scoped-node' },
            { nodeIndex: 3, nodeClass: 'scoped-node' },
          ])
          expectDependencyClasses([
            { dependencyIndex: 1, dependencyClass: 'direct-dependency' },
            { dependencyIndex: 2, dependencyClass: 'scoped-dependency' },
            { dependencyIndex: 3, dependencyClass: 'scoped-dependency' },
          ])
        })

        describe('and the mouse hovers away from the first node', () => {
          beforeEach(async () => {
            await createWrapper()
            await wrapper
              .get('#component-dependency-graph-node-0')
              .trigger('mouseout')
          })

          test('renders all the connections as scoped connections', () => {
            expect(wrapper.html()).toMatchSnapshot()
            expectNodeClasses([
              { nodeIndex: 0, nodeClass: 'scoped-node' },
              { nodeIndex: 1, nodeClass: 'scoped-node' },
              { nodeIndex: 2, nodeClass: 'scoped-node' },
              { nodeIndex: 3, nodeClass: 'scoped-node' },
            ])
            expectDependencyClasses([
              { dependencyIndex: 1, dependencyClass: 'scoped-dependency' },
              { dependencyIndex: 2, dependencyClass: 'scoped-dependency' },
              { dependencyIndex: 3, dependencyClass: 'scoped-dependency' },
            ])
          })
        })
      })
    })
  })
})
