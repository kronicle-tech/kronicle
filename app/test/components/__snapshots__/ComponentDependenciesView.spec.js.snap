// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentDependenciesView when components prop is set to an empty array renders no dependencies 1`] = `
"<div>
  <transition-stub name=\\"\\" css=\\"true\\" enterclass=\\"\\" leaveclass=\\"show\\" entertoclass=\\"show\\" leavetoclass=\\"\\" enteractiveclass=\\"\\" leaveactiveclass=\\"\\">
    <div role=\\"alert\\" aria-live=\\"polite\\" aria-atomic=\\"true\\" class=\\"alert alert-dismissible alert-info\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"close\\">Ã—</button>
      Click a dot in the dependencies diagram to see more information about that
      component
    </div>
  </transition-stub>
  <!---->
  <div class=\\"graph\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" overflow=\\"auto\\" width=\\"100%\\" height=\\"1200\\" id=\\"component-dependency-graph\\">
      <rect fill-opacity=\\"0\\" class=\\"component-dependency-graph-background\\"></rect>
      <defs>
        <marker id=\\"component-dependency-graph-dependency-marker-all\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker all-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
        <marker id=\\"component-dependency-graph-dependency-marker-manual\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker manual-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
        <marker id=\\"component-dependency-graph-dependency-marker-scope-related\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker scope-related-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
        <marker id=\\"component-dependency-graph-dependency-marker-scoped\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker scoped-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
        <marker id=\\"component-dependency-graph-dependency-marker-related\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker related-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
        <marker id=\\"component-dependency-graph-dependency-marker-direct\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker direct-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
        <marker id=\\"component-dependency-graph-dependency-marker-selected\\" orient=\\"auto-start-reverse\\" viewBox=\\"0 0 7.1 11.5\\" markerWidth=\\"7.1\\" markerHeight=\\"11.5\\" markerUnits=\\"userSpaceOnUse\\" refX=\\"20\\" refY=\\"5.75\\" class=\\"dependency-marker selected-dependency-marker\\">
          <path d=\\"M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5\\"></path>
        </marker>
      </defs>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
      <g></g>
    </svg></div>
  <div tabindex=\\"-1\\" class=\\"b-sidebar-outer\\">
    <!---->
    <transition-stub css=\\"true\\" enterclass=\\"\\" leaveclass=\\"show\\" entertoclass=\\"show\\" leavetoclass=\\"\\" enteractiveclass=\\"slide\\" leaveactiveclass=\\"slide\\">
      <div id=\\"component\\" tabindex=\\"-1\\" role=\\"dialog\\" aria-modal=\\"true\\" aria-hidden=\\"true\\" class=\\"b-sidebar b-sidebar-right bg-dark text-light\\" style=\\"width: 600px; display: none;\\">
        <header class=\\"b-sidebar-header\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"close text-light\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1em\\" height=\\"1em\\" focusable=\\"false\\" role=\\"img\\" aria-label=\\"x\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"currentColor\\" class=\\"bi-x b-icon bi\\">
              <g>
                <path d=\\"M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z\\"></path>
              </g>
            </svg></button><span></span></header>
        <div class=\\"b-sidebar-body\\">
          <div class=\\"m-3\\">
            <!---->
          </div>
        </div>
        <!---->
      </div>
    </transition-stub>
    <!---->
    <transition-stub name=\\"\\" css=\\"true\\" enterclass=\\"\\" leaveclass=\\"show\\" entertoclass=\\"show\\" leavetoclass=\\"\\" enteractiveclass=\\"fade\\" leaveactiveclass=\\"fade\\">
      <div class=\\"b-sidebar-backdrop bg-dark\\" style=\\"display: none;\\"></div>
    </transition-stub>
  </div>
</div>"
`;
